plugin/cache/setup_test.go

```diff
@@ -9,72 +9,101 @@ import (
 
 func TestSetup(t *testing.T) {
 	tests := []struct {
-		input            string
-		shouldErr        bool
-		expectedNcap     int
-		expectedPcap     int
-		expectedNttl     time.Duration
-		expectedMinNttl  time.Duration
-		expectedPttl     time.Duration
-		expectedMinPttl  time.Duration
-		expectedPrefetch int
+		input                string
+		shouldErr            bool
+		expectedNcap         int
+		expectedPcap         int
+		expectedNttl         time.Duration
+		expectedMinNttl      time.Duration
+		expectedPttl         time.Duration
+		expectedMinPttl      time.Duration
+		expectedPrefetch     int
+		expectedServeExpired bool
```

Comment:
just put these checks in a small new test that just checks the new elements. Now we just have a giant list of `false` and a few `true`'s that check the new setup code. A new selfcontained small test is almost certainly better.

Reply:
Done. A lot cleaner now.